lib/cli/src/commands/run/wasi.rs

```diff
@@ -105,6 +107,35 @@ pub struct Wasi {
     #[clap(long = "enable-async-threads")]
     pub enable_async_threads: bool,
 
+    /// Specifies the snapshot file that Wasmer will use to store
+    /// the state of the WASM process so that it can be later restored
+    #[cfg(feature = "snapshot")]
+    #[clap(long = "snapshot-to")]
+    pub snapshot_to: Option<PathBuf>,
+
+    /// Indicates what events will cause a snapshot to be taken
+    /// and written to the snapshot file.
+    ///
+    /// If not specified, the default is to snapshot on idle plus if a
+    /// snapshot period is provided it will also default to periodic snapshots
+    /// as well.
+    #[cfg(feature = "snapshot")]
+    #[clap(long = "snapshot-on")]
+    pub snapshot_on: Vec<SnapshotTrigger>,
+
+    /// Adds a timer (measured in seconds) that takes snapshots of the process and dumps the
+    /// journal of events to the snapshot file. When specifying this parameter it implies
+    /// that `--snapshot-on timer` has also been specified.
+    #[cfg(feature = "snapshot")]
+    #[clap(long = "snapshot-timer")]
+    pub snapshot_timer: Option<u64>,
+
+    /// When specified, the runtime will restore a previous snapshot
+    /// using the supplied file.
+    #[cfg(feature = "snapshot")]
+    #[clap(long = "resume-from")]
+    pub resume_from: Option<PathBuf>,
```

Comment:
This doesn't exist anymore anyway, instead what we did was modify the CLI to use the `--journal` parameter which where former arguments are read-only applied to the state and the later (or last) journal is first applied then appended to. If one wants to do the use case of a delta then only need specify an empty (or non-existent file) as the last journal and it will all stay immutable which is what this argument above you have referenced was designed to cover.